src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,829 @@
+                .OrderByDescending(reference => reference.Location.SourceSpan.Start);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+            // Adding the original document to account for refactorings that do not have any invocations
+            methodCallSitesBuilder.Add(document, new ArrayBuilder<TInvocationExpressionSyntax>().ToImmutable());
+
+            foreach (var refLocation in referencedLocations)
+            {
+                // Does not support cross-language references currently
+                if (refLocation.Document.Project.Language == document.Project.Language)
+                {
+                    var invocation = refLocation.Location.FindNode(cancellationToken).GetRequiredParent();
+
+                    // Only adding items that are of type InvocationExpressionSyntax
+                    // TODO: in future, add annotation to items that aren't InvocationExpressionSyntax
+                    if (!methodCallSitesBuilder.ContainsKey(refLocation.Document))
+                    {
+                        var invocations = new ArrayBuilder<TInvocationExpressionSyntax>();
+                        if (syntaxFacts.IsInvocationExpression(invocation))
+                        {
+                            invocations.Add((TInvocationExpressionSyntax)invocation);
+                            methodCallSitesBuilder.Add(refLocation.Document, invocations.ToImmutable());
+                        }
+                    }
+                    else
+                    {
+                        if (syntaxFacts.IsInvocationExpression(invocation))
+                        {
+                            var invocations = methodCallSitesBuilder[refLocation.Document];
+                            invocations = invocations.Add((TInvocationExpressionSyntax)invocation);
+                            methodCallSitesBuilder[refLocation.Document] = invocations;
+                        }
+                    }
+                }
+            }
+            return methodCallSitesBuilder.ToImmutable();
+        }
+
+        /// <summary>
+        /// If the parameter is optional and the invocation does not specify the parameter, then
+        /// a named argument needs to be introduced.
+        /// </summary>
+        private static SeparatedSyntaxList<SyntaxNode> AddArgumentToArgumentList(
+            SeparatedSyntaxList<SyntaxNode> invocationArguments, SyntaxGenerator generator,
+            SyntaxNode newArgumentExpression, int insertionIndex, string name, bool named)
+        {
+            var argument = named
+                ? generator.Argument(name, RefKind.None,
+                    newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation))
+                :
+                generator.Argument(newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
```

Comment:
```suggestion
            var argument = named
                ? generator.Argument(name, RefKind.None, newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation))
                : generator.Argument(newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
```

Reply:
also remove the annotation here, and just move it to below line.

Reply:
added annotation